lib/cellect/server/workflow.rb

```diff
@@ -41,23 +46,39 @@ def initialize(name, pairwise: false, prioritized: false)
         self.users = { }
         self.pairwise = !!pairwise
         self.prioritized = !!prioritized
-        self.subjects = set_klass.new
+        self.subjects ||= set_klass.new
+        self.state = :initializing
+        self.can_reload = false
       end
 
       # Loads subjects from the adapter
       def load_data
-        self.state = :initializing
-        self.subjects = set_klass.new
-        Cellect::Server.adapter.load_data_for(name).each do |hash|
-          subjects.add hash['id'], hash['priority']
-        end
+        return if self.state == :ready
+        self.state = :loading
+        load_adapter_data(self.subjects)
+        reset_can_reload_timer
```

Comment:
I wonder if this timer thing is better than setting `can_reload_from` to `RELOAD_TIMEOUT.from_now` and then testing if it's in the past inside `can_reload_data?`. Less moving parts, since there's no separate timer to maintain (no `finalize` either). 

Up to you.